### DISCUSSION TITLE
Allow more redirection options when using /admin/login?redirect={anything}

### DISCUSSION THREAD

**Discussion by cube-dan** [directus/directus#16644] [Feature Requests]
### Describe the Bug

When using `res.redirect(301,'/admin/login?redirect=/extension-name/some-path')` inside a custom endpoint, the user is directed to a url with the /admin prefix added to the front of the url.

This happens regardless of what you prefix the extension's route with. For example: 
`http://0.0.0.0:8055/admin/login?redirect=/extension-name/some-path`
will spit you out on:
`http://0.0.0.0:8055/admin/extension-name/some-path`

Supplying a protocol relative url won’t work either:
`http://0.0.0.0:8055/admin/login?redirect=//extension-name/some-path`
will spit you out on:
`http://0.0.0.0:8055/admin/http://0.0.0.0:8055/extension-name/some-path`

Specifying the full url doesn’t help. For example:
`http://0.0.0.0:8055/admin/login?redirect=http://0.0.0.0:8055/extension-name/some-path`
will spit you out on:
`http://0.0.0.0:8055/admin/http://0.0.0.0:8055/extension-name/some-path`

### To Reproduce

1. Have a directus instance already ready already; [I'm using self hosted](https://docs.directus.io/self-hosted/quickstart.html#_1-installation)
2. Shutdown your directus install.
3. Create a directus extension [using the cli](https://docs.directus.io/extensions/creating-extensions.html#building-your-extension); make it an [endpoint](https://docs.directus.io/extensions/endpoints.html#extension-entrypoint).
4. Add this code to your extension's index.js file
```
export default {
	id: 'extension-name',
	handler: (router) => {
		router.get('/some-path', (req, res) => {
			res.send("You'll see a 404 instead of this content"));
        	}
		router.get('/another-path', (req, res) => {
			res.redirect(301,`/admin/login?redirect=/extension-name/some-path`);
        	}
	},
};
```
5. Package your [local extension](https://docs.directus.io/extensions/creating-extensions.html#local-extension) then install it locally.
6. Startup directus
7. While logged out of directus, visit `http://0.0.0.0:8055/extension-name/another-path`. You'll be redirected to the admin login page with `?redirect=/extension-name/some-path` appended to the `/admin/login` url, like this: `http://0.0.0.0:8055/admin/login?redirect=/extension-name/some-path`
8. After logging into directus you'll be plopped out on a 404 page, this one: `http://0.0.0.0:8055/admin/extension-name/some-path` instead of where the extension intended to send you: `http://0.0.0.0:8055/extension-name/some-path`

... thanks for taking a look at this guys. 

-Dan

### Errors Shown

Does the 404 count?

### What version of Directus are you using?

v9.21.0

### What version of Node.js are you using?

v14.20.1

### What database are you using?

10.8.3-MariaDB Homebrew

### What browser are you using?

Version 107.0.5304.87 (Official Build) (x86_64)

### How are you deploying Directus?

running locally, self-hosted, @ http://localhost:8055

**Comment by br41nslug** [comment]
The login redirect is not an unlimited redirect and only allows for redirection to routes known by the App currently thus `/admin/*` urls only.

**Comment by cube-dan** [comment]
@br41nslug,

Thanks for your reply, but I don’t think I understand what you mean by "only allows for redirection to routes known by the App". Are urls defined by an endpoint extension unknown to the App?

-Dan

**Reply by br41nslug** [reply]
What i mean is that the redirect is handed over to the App frontend router which can only route to paths it knows (it is not doing a blind `window.location = url` redirect. The advantage of this approach is security but the drawback is that backend routes (API, custom endpoints) are not known to the frontend router.

https://github.com/directus/directus/blob/main/app/src/routes/login/components/login-form/login-form.vue#L117

**Comment by cube-dan** [comment]
@br41nslug,

Is there a "proper Directus" way to make backend routes (API, custom endpoints) known to the frontend router so a users can be redirected from the Directus login page back to a custom API endpoint?

-Dan

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
